File: MyProjects/MAP/Assignment 7/Interpreter/Repository/IRepo.cs
using System;
using System.Collections.Generic;
using System.Text;
using Interpreter.Model;

namespace Interpreter.Repository
{
    public interface IRepo
    {
        void addPrg(ProgramState prg);
        ProgramState getCurrentPrg();
        void logProgramStateExec(string logFilePath);
    }
}

File: MyProjects/MAP/Practical Exam/Sleep/src/Repository/IRepo.java
package Repository;

import Model.ProgramState;
import java.io.IOException;
import java.util.List;

public interface IRepo {
    void addPrg(ProgramState prg);
    void logProgramStateExec(ProgramState state) throws IOException;
    List<ProgramState> getPrgList();
    void setPrgList(List<ProgramState> list);
}
